iap/validate_jwt.py

```diff
@@ -27,18 +27,41 @@
 import requests
 
 
-def validate_iap_jwt(base_url, iap_jwt):
+def validate_iap_jwt(cloud_project_number, iap_jwt,
+                     app_engine_project_id=None,
+                     compute_engine_backend_service_id=None):
     """Validate a JWT passed to your application by Identity-Aware Proxy.
 
+    One and only of of (app_engine_project_id, compute_engine_backend_service_id)
+    must be set.
+
     Args:
-      base_url: The URL from the incoming request, minus any path, query, etc.
-                For instance: "https://example.com:8443" or
-                "https://example.appspot.com" .
-      iap_jwt: The contents of the X-Goog-Authenticated-User-JWT header.
+      cloud_project_number: The project *number* for your Google Cloud project.
+          This is returned by 'gcloud projects describe $PROJECT_ID', or
+          in the Project Info card in Cloud Console.
+      iap_jwt: The contents of the X-Goog-IAP-JWT-Assertion header.
+      app_engine_project_id: For App Engine resources, this must be set to
+          your cloud project ID.
+      compute_engine_backend_service_id: For Compute Engine and Container Engine
+          resources this must be set to the ID of the backend service used to
+          access the application. See
+          https://cloud.google.com/iap/docs/signed-headers-howto for details on
+          how to get this value.
 
     Returns:
       (user_id, user_email, error_str).
     """
+    if not (bool(app_engine_project_id) ^ bool(compute_engine_backend_service_id)):
```

Comment:
This is a code smell, can you maybe split this into a couple of functions like `validate_iap_jwt_from_compute` that call a common helper with a `target_audience` or alternatively have one helper method that figures out the target audience and one to validate the jwt?